org.eclipse.xtext.builder/src/org/eclipse/xtext/builder/preferences/BuilderPreferenceAccess.java

```diff
@@ -32,6 +32,36 @@
 	 * Name of a preference for configuring whether the builder participant is enabled or not.
 	 */
 	public static final String PREF_AUTO_BUILDING = "autobuilding"; //$NON-NLS-1$
+	/**
+	 * Name of a preference defining the scheduling rule strategy for the builder.
+	 * @since 2.16
+	 */
+	public static final String PREF_SCHEDULING_RULE = "schedulingrule"; //$NON-NLS-1$
+	/**
+	 * Options for scheduling rules.
+	 * @since 2.16
+	 */
+	public static enum SchedulingOption {
+		WORKSPACE ("workspace"),
+		// all Xtext projects in the workspace
+		XTEXT_PROJECTS ("all_xtext_projects"), //$NON-NLS-1$
+		// current built project
+		PROJECT ("project");
+		private String name;
+		
+		SchedulingOption (String name) {
+			this.name = name;
+		}
+		
+		static SchedulingOption fromName (String name) {
+			for (SchedulingOption opt: values()) {
+				if (opt.name.equals(name)) {
+					return opt;
+				}
+			}
+			throw new IllegalArgumentException(name);
```

Comment:
log a warning and default?

Reply:
Would be a really bad programming error if you'd try to get a value for a non existing name

Reply:
Why not `SchedulingOption.valueOf(..)`?

Reply:
Using valueOf() now. I had a mix of upper/lower case before, that was the reason. Comment is obsolete now.